src/engine/bank05.asm

```diff
@@ -54,31 +54,1059 @@ PointerTable_14000: ; 14000 (05:4000)
+;	- card ID is Eevee and a corresponds to an energy type
+;	  of water, fire or lightning;
+;	- type of card in register a is the same as wTempCardType.
+; used for knowing if a given energy card can be discarded
+; from a given Pokémon card
+; input:
+;	a = energy card attached to Pokémon to check
+;	[wTempCardType] = TYPE_ENERGY_* of given Pokémon
+;	[wTempCardID] = card index of Pokémon card to check
+CheckIfEnergyIsUseful: ; 14184 (5:4184)
+	push de
+	call GetCardIDFromDeckIndex
+	ld a, e
+	cp DOUBLE_COLORLESS_ENERGY
+	jr z, .set_carry
+	ld a, [wTempCardType]
+	cp TYPE_ENERGY_DOUBLE_COLORLESS
+	jr z, .set_carry
+	ld a, [wTempCardID]
+
+	ld d, PSYCHIC_ENERGY
+	cp EXEGGCUTE
+	jr z, .check_energy
+	cp EXEGGUTOR
+	jr z, .check_energy
+	cp PSYDUCK
+	jr z, .check_energy
+	cp GOLDUCK
+	jr z, .check_energy
+
+	ld d, WATER_ENERGY
+	cp SURFING_PIKACHU1
+	jr z, .check_energy
+	cp SURFING_PIKACHU2
+	jr z, .check_energy
+
+	cp EEVEE
+	jr nz, .check_type
+	ld a, e
+	cp WATER_ENERGY
+	jr z, .set_carry
+	cp FIRE_ENERGY
+	jr z, .set_carry
+	cp LIGHTNING_ENERGY
+	jr z, .set_carry
+
+.check_type
+	ld d, $00 ; unnecessary?
+	call GetCardType
+	ld d, a
+	ld a, [wTempCardType]
+	cp d
+	jr z, .set_carry
+	pop de
+	or a
+	ret
+	
+.check_energy
+	ld a, d
+	cp e
+	jr nz, .check_type
+.set_carry
+	pop de
+	scf
+	ret
+; 0x141da
+
+Func_141da: ; 141da (5:41da)
+	INCROM $141da, $14226
 
 Func_14226: ; 14226 (5:4226)
 	call CreateHandCardList
 	ld hl, wDuelTempList
-.check_for_next_pokemon
+.check_for_next_card
 	ld a, [hli]
 	ldh [hTempCardIndex_ff98], a
 	cp $ff
 	ret z
+
 	call LoadCardDataToBuffer1_FromDeckIndex
 	ld a, [wLoadedCard1Type]
 	cp TYPE_ENERGY
-	jr nc, .check_for_next_pokemon
+	jr nc, .check_for_next_card
 	ld a, [wLoadedCard1Stage]
 	or a
-	jr nz, .check_for_next_pokemon
+	jr nz, .check_for_next_card
 	push hl
 	ldh a, [hTempCardIndex_ff98]
 	call PutHandPokemonCardInPlayArea
 	pop hl
-	jr .check_for_next_pokemon
+	jr .check_for_next_card
 ; 0x1424b
 
-	INCROM $1424b, $14663
+; returns carry if Pokémon at hTempPlayAreaLocation_ff9d
+; can't use a move or if that selected move doesn't have enough energy
+; input:
+; 	[hTempPlayAreaLocation_ff9d] = location of Pokémon card
+;	[wSelectedMoveIndex]		 = selected move to examine
+CheckIfCardCanUseSelectedMove: ; 1424b (5:424b)
+	ldh a, [hTempPlayAreaLocation_ff9d]
+	or a
+	jr nz, .bench
+
+	bank1call HandleCantAttackSubstatus
+	ret c
+	bank1call CheckIfActiveCardParalyzedOrAsleep
+	ret c
+
+	ld a, DUELVARS_ARENA_CARD
+	call GetTurnDuelistVariable
+	ld d, a
+	ld a, [wSelectedMoveIndex]
+	ld e, a
+	call CopyMoveDataAndDamage_FromDeckIndex
+	call HandleAmnesiaSubstatus
+	ret c
+	ld a, EFFECTCMDTYPE_INITIAL_EFFECT_1
+	call TryExecuteEffectCommandFunction
+	ret c
+	
+.bench
+	call CheckEnergyNeededForAttack
+	ret c ; can't be used
+	ld a, $0d ; $00001101
+	call CheckLoadedMoveFlag
+	ret
+; 0x14279
+
+; load selected move from Pokémon in hTempPlayAreaLocation_ff9d
```

Comment:
I would avoid the use of tabs for formatting comments in general. The leading tab to list input or output arguments is fine, but for everything else I'd just use a bunch of spaces as you see fit.